publication-commons/src/test/java/no/unit/nva/publication/model/business/publicationchannel/PublicationChannelTest.java

```diff
@@ -170,13 +175,74 @@ void shouldThrowExceptionWhenIdentityServiceRespondsWithUnhandledError()
     }
 
     @Test
-    void shouldReturnListingResultWhenListingPublicationChannels()
-        throws NotFoundException {
+    void shouldUpdatePublicationChannelWhenPublisherIsUpdated() throws BadRequestException, NotFoundException {
+        var publisherId = randomPublisherId(randomUUID());
+        var persistedPublication = persistDegreeWithPublisher(new Publisher(publisherId));
+
+        var existingResource = resourceService.getResourceByIdentifier(persistedPublication.getIdentifier());
+        var existingPublicationChannel = existingResource.getPublicationChannels().getFirst();
+        assertEquals(getChannelClaimIdentifier(publisherId), existingPublicationChannel.getIdentifier().toString());
+
+        var newChannelIdentifier = randomUUID();
+        var newPublisherId = randomPublisherId(newChannelIdentifier);
+        persistedPublication.getEntityDescription().getReference().setPublicationContext(degreeWithPublisherId(newPublisherId));
+
+        when(identityService.getChannelClaim(toChannelClaimUri(newChannelIdentifier))).thenThrow(new NotFoundException(randomString()));
+        resourceService.updatePublication(persistedPublication);
+
+        var updatedResource = resourceService.getResourceByIdentifier(persistedPublication.getIdentifier());
+        var updatedPublicationChannels = updatedResource.getPublicationChannels();
+
+        assertEquals(ONE, updatedPublicationChannels.size());
+        assertEquals(getChannelClaimIdentifier(newPublisherId),
+                     updatedPublicationChannels.getFirst().getIdentifier().toString());
+    }
+
+    @Test
+    void shouldRemovePublicationChannelWhenPublisherIsRemoved() throws BadRequestException, NotFoundException {
+        var publisherId = randomPublisherId(randomUUID());
+        var persistedPublication = persistDegreeWithPublisher(new Publisher(publisherId));
+
+        var existingResource = resourceService.getResourceByIdentifier(persistedPublication.getIdentifier());
+        assertFalse(existingResource.getPublicationChannels().isEmpty());
+
+        persistedPublication.getEntityDescription().getReference().setPublicationContext(degreeWithPublisher(null));
+        resourceService.updatePublication(persistedPublication);
+
+        var updatedResource = resourceService.getResourceByIdentifier(persistedPublication.getIdentifier());
+        assertTrue(updatedResource.getPublicationChannels().isEmpty());
+    }
+
+    @Test
+    void shouldAddPublicationChannelWhenNoPublisherIsAdded() throws BadRequestException, NotFoundException {
```

Comment:
I think test name is wrong. Should be something like `shouldPersistPublicationChannelWhenPublisherIsSetOnPublication()`